adapter/internal/operator/synchronizer/data_store.go

```diff
@@ -85,6 +85,10 @@ func (ods *OperatorDataStore) processAPIState(apiNamespacedName types.Namespaced
 			events = append(events, routeEvents...)
 		}
 	} else {
+		if cachedAPI.ProdHTTPRoute != nil {
+			updated = true
+			events = append(events, "Production")
+		}
 		cachedAPI.ProdHTTPRoute = nil
```

Comment:
because of this line, the fix won't work right? i guess it should be inside a else block

Reply:
We need to make it ni right? The fix is working as the added cucumber test is passing now i tested it manually also